test/extended/networking/cnimigration.go

```diff
@@ -0,0 +1,340 @@
+	if err := retry.RetryOnConflict(wait.Backoff{Steps: 10, Duration: time.Second}, func() error {
+		_, err := c.ConfigV1().Networks().Patch(context.Background(), networkCRName, kubetypes.MergePatchType, patch, metav1.PatchOptions{})
+		return err
+	}); err != nil {
+		return err
+	}
+	return nil
+}
+
+func isMigrationInProgressTrue(c configv1client.Interface) (bool, error) {
+	clusterConfig, err := c.ConfigV1().Networks().Get(context.TODO(), networkCRName, metav1.GetOptions{})
+	if err != nil {
+		return false, err
+	}
+	if meta.IsStatusConditionPresentAndEqual(clusterConfig.Status.Conditions, networkopernames.NetworkTypeMigrationInProgress, metav1.ConditionTrue) {
+		return true, nil
+	}
+	return false, nil
+}
+
+func isMigrationComplete(c configv1client.Interface) (bool, error) {
+	clusterConfig, err := c.ConfigV1().Networks().Get(context.TODO(), networkCRName, metav1.GetOptions{})
+	if err != nil {
+		return false, err
+	}
+	if !meta.IsStatusConditionPresentAndEqual(clusterConfig.Status.Conditions, networkopernames.NetworkTypeMigrationInProgress, metav1.ConditionFalse) {
+		return false, nil
+	}
+	if !meta.IsStatusConditionPresentAndEqual(clusterConfig.Status.Conditions, networkopernames.NetworkTypeMigrationMTUReady, metav1.ConditionUnknown) {
+		return false, nil
+	}
+	if !meta.IsStatusConditionPresentAndEqual(clusterConfig.Status.Conditions, networkopernames.NetworkTypeMigrationOriginalCNIPurged, metav1.ConditionUnknown) {
+		return false, nil
+	}
+	if !meta.IsStatusConditionPresentAndEqual(clusterConfig.Status.Conditions, networkopernames.NetworkTypeMigrationTargetCNIInUse, metav1.ConditionUnknown) {
+		return false, nil
+	}
+	if !meta.IsStatusConditionPresentAndEqual(clusterConfig.Status.Conditions, networkopernames.NetworkTypeMigrationTargetCNIAvailable, metav1.ConditionUnknown) {
+		return false, nil
+	}
+	return true, nil
+}
+
+func isCNIDeployed(c configv1client.Interface, cni string) (bool, error) {
+	clusterConfig, err := c.ConfigV1().Networks().Get(context.TODO(), networkCRName, metav1.GetOptions{})
+	if err != nil {
+		return false, err
+	}
+	if clusterConfig.Status.NetworkType == cni {
+		return true, nil
+	}
+	return false, nil
+}
+
+func getCurrentCNI(configClient configv1client.Interface) (string, error) {
+	clusterConfig, err := configClient.ConfigV1().Networks().Get(context.TODO(), networkCRName, metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+	return clusterConfig.Status.NetworkType, nil
+}
+
+func migrateCNI(c configv1client.Interface, config *restclient.Config, targetCNI string) error {
+	if inProgress, err := isMigrationInProgressTrue(c); err != nil {
+		return err
+	} else if inProgress {
+		return fmt.Errorf("migration is already in-progress")
+	}
+	if isDeployed, err := isCNIDeployed(c, targetCNI); err != nil {
+		return err
+	} else if isDeployed {
+		return fmt.Errorf("CNI %q is already deployed", targetCNI)
+	}
+	if err := enableLiveMigrationFeatureGate(c); err != nil {
```

Comment:
Can we create the test cluster with techpreview enabled instead of enabling it here?

Reply:
Yes we can. I did this in https://github.com/openshift/release/pull/46805
I removed enabling the feature gate here as its not required and must be a prerequisite now. This will help us make less changes in the future also.